tests/Games/NexusMods.Games.StardewValley.Tests/DependencyDiagnosticEmitterTests.cs

```diff
@@ -102,6 +117,19 @@ public async Task Test_RequiredDependencyIsOutdated()
         var dependent = requiredDependencyIsOutdatedMessageData.Dependent.ResolveData(ServiceProvider, Connection);
         dependent.AsLoadoutItem().ParentId.Should().Be(farmTypeManager.LoadoutItemGroupId, because: "Farm Type Manager is the dependent");
 
+        // Test disabled cases
+        await DisableItem(farmTypeManager);
+        
+        (await GetAllDiagnostics(loadout)).OfType<Diagnostic<Diagnostics.RequiredDependencyIsOutdatedMessageData>>()
+            .Should().BeEmpty(because: "The dependant Farm Type Manager is disabled");
```

Comment:
This should just be `ShouldHaveNoDiagnostics`.

Reply:
That wouldn't work because you actually get the `DisabledRequiredDependencyMessageData` instead

Reply:
Right, then you should assert that you got a single diagnostic with `DisabledRequiredDependencyMessageData` and no diagnostics with `RequiredDependencyIsOutdatedMessageData`.

Reply:
I think we should get `MissingRequiredDependency` instead tbh, basically check that the disabled dependency actually matches min required version.